### PR TITLE
Downgrade Java for client libraries to 8

### LINKED ISSUE #33051: Handle keycloak repository
See parent issue https://github.com/keycloak/keycloak/issues/33050 for the details of what should be probably done in keycloak repository.

### PR DESCRIPTION
**Pull request by rmartinc**
Closes #33051

Downgrade for java 8 in all the client packages. Minor modifications in classes to make them compatible with jdk-8. The only weird thing is `ci.yml` is modified because using `test` did not incorporate the multi-release and Base UT test failed with not supported exception. My feeling is that using `test` instead of `install` incorporates classes from the the `target/classes` directory and the multi-release is not detected. Using `install` incoporporate the jars which are perfectly fine and using the multi-release. Don't know why this didn't happen before, maybe different versions of maven.

### CODE REVIEW

File: authz/client/pom.xml [hunk 1]
```diff
@@ -18,6 +18,12 @@
     <description>Keycloak Authz: Client API. This module is supposed to be used just in the Keycloak repository for the testsuite. It is NOT supposed to be used by the 3rd party applications.
         For the use by 3rd party applications, please use `org.keycloak:keycloak-authz-client` module.</description>
 
+    <properties>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>8</maven.compiler.target>
+        <maven.compiler.release>8</maven.compiler.release>
```
Comment (jonkoops) [inline authz/client/pom.xml:24]: If I recall correctly setting the `release` should be enough to get compatible byte-code for a certain API level. There should be no need to set the `source` to `8`.

So best case scenario:
```suggestion
        <maven.compiler.release>8</maven.compiler.release>
```

And at worst:
```suggestion
        <maven.compiler.target>8</maven.compiler.target>
        <maven.compiler.release>8</maven.compiler.release>
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: authz/client/pom.xml (hunk 1 above)
Comment (rmartinc) [reply authz/client/pom.xml]: yes, you are right. It's done for the four poms now.

**Review by graziang — APPROVED** [review]
@rmartinc LGTM!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -85,7 +85,7 @@ jobs:
             SEP=","
           done
 
-          ./mvnw test -pl "$PROJECTS" -am
+          ./mvnw install -pl "$PROJECTS" -am
```
Comment (jonkoops) [inline .github/workflows/ci.yml:88]: Did this command have to be replaced because of the the downgrade, or is there another motivation?

**Review by jonkoops — DISMISSED** [review]
Because the `source` has been reverted most of the syntax changes (e.g. `var`) could also be reverted to what they were before this PR. But the changes here do no harm, so LGTM.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (rmartinc) [reply .github/workflows/ci.yml:88]: Yes, it's explained in the description of the PR. It seems that doing `test` the multi-release is not detected. My feeling is that the classpath when testing is using the `core/target/classes` (to access core classes) which has no `MANIFEST.MF` and it is not multi-release. Doing the `install` it uses the jar which is OK. I don't know how this worked before, probably a different maven version now.

Comment (jonkoops) [reply .github/workflows/ci.yml:88]: Gotcha, thanks for the explainer.

**Comment by rmartinc** [comment]
> Because the `source` has been reverted most of the syntax changes (e.g. `var`) could also be reverted to what they were before this PR. But the changes here do no harm, so LGTM.

No, I think that `var` will not work. Don't know about other things, but for sure `var` does not work when using release to 8.

**Comment by jonkoops** [comment]
> No, I think that `var` will not work. Don't know about other things, but for sure `var` does not work when using release to 8.

Strange, my IntelliJ doesn't complain about it, but it indeed does not compile.

**Comment by rmartinc** [comment]
:smile: Then maybe it's better to have the source set to 8 too.

**Comment by rmartinc** [comment]
Adding back source and target to allow IDEs to detect correctly the version.

**Review by mposolda — APPROVED** [review]
@rmartinc @graziang @jonkoops Thanks for the PR and the review!

**Review by jonkoops — APPROVED** [review]

**Pull request merged**